MantidQt/MantidWidgets/src/InstrumentView/Shape2DCollection.cpp

```diff
@@ -685,7 +685,9 @@ void Shape2DCollection::loadFromTableWorkspace(
   for (size_t i = 0; i < ws->rowCount(); ++i) {
     const auto params = col[i];
     auto shape = Shape2D::loadFromProject(params);
-    m_shapes.append(shape);
+    if (shape != nullptr) {
```

Comment:
My thinking for doing it was:
1. According to mantid C++ coding standards: *Integers should not be cast to Booleans.* Raw pointers can be considered integers.
2. `auto shape` hides the actual type of variable `shape`. Comparing it with `nullptr` shows its type clearly, no need to investigate further what `if (shape)` means.